syntax = "proto3";

package grpc.health.v1;

option java_package = "grpc.health.v1";
option java_multiple_files = true;
option java_outer_classname = "HealthProto";

/**
 * Standard gRPC health checking protocol.
 * This is based on the official gRPC health checking protocol:
 * https://github.com/grpc/grpc/blob/master/doc/health-checking.md
 */
service Health {
  // Performs a health check
  rpc Check(HealthCheckRequest) returns (HealthCheckResponse);
  
  // Performs a watch for the serving status of the requested service
  rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse);
}

message HealthCheckRequest {
  // Service name to check. Empty string means check the overall server health.
  string service = 1;
}

message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
    SERVICE_UNKNOWN = 3;  // Used only by the Watch method.
  }
  ServingStatus status = 1;
}